### PR TITLE
LRQA-70504, LRQA-70572 | master

### PR DESCRIPTION
**Pull request by Kvale1733**
https://issues.liferay.com/browse/LRQA-70504
https://issues.liferay.com/browse/LRQA-70572

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:security:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase [hunk 1]
```diff
@@ -25,21 +25,25 @@ definition {
 		}
 	}
 
-	@description = "This is a use case for LPS-90727."
+	@description = "This is a use case for LPS-90727. Assert no XSS vulnerability when modifying a specific URL from the control panel"
 	@priority = "5"
 	test AssertXSSVulnerabilityInControlPanel {
-		var portalURL = PropsUtil.get("portal.url");
+		task ("Go to Account Settings and save p_p_auth param from the URL") {
+			var portalURL = PropsUtil.get("portal.url");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase:32]: Is it localhost? Maybe we could remove this and hardcode the URL in the steps below?

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase [hunk 2]
```diff
@@ -25,21 +25,25 @@ definition {
...
 
-		var currentURL = Navigator.getCurrentURL();
-		var p_p_authParam = StringUtil.extractLast("${currentURL}", "=");
+			var currentURL = Navigator.getCurrentURL();
+			var p_p_authParam = StringUtil.extractLast("${currentURL}", "=");
+		}
 
-		Navigator.openSpecificURL(url = "${portalURL}/en/group/control_panel/manage;'*prompt()*'?p_p_id=com_liferay_my_account_web_portlet_MyAccountPortlet&p_p_lifecycle=0&p_p_state=maximized&p_p_auth=${p_p_authParam}");
+		task ("Go to url with p_p_authParam and assert no dialog box appears") {
+			Navigator.openSpecificURL(url = "${portalURL}/en/group/control_panel/manage;'*prompt()*'?p_p_id=com_liferay_my_account_web_portlet_MyAccountPortlet&p_p_lifecycle=0&p_p_state=maximized&p_p_auth=${p_p_authParam}");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase:43]: Maybe this param could be hardcoded?

Comment (Kvale1733) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase:43]: Hey Gabor it looks like this param is generated with each portal instance so I don't think it should be hardcoded.

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:security:nocompile

**Comment by Kvale1733** [comment]
ci:test:security

**Comment by Kvale1733** [comment]
ci:stop

**Comment by Kvale1733** [comment]
ci:stop

**Comment by Kvale1733** [comment]
ci:test:security:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase (hunk 2 above)
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase]: Ok

**Comment by gaborlovasdotcom** [comment]
Hi @Kvale1733!

For ValidateXSSVulnerabilities

we should keep the 

`var portalURL = PropsUtil.get("portal.url");

			var maliciousURL1 = '''
					${portalURL}`
					
formula, and no hard code, due we don't know what Portal host will be used, as you mentioned in one of your comments

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase [hunk 3]
```diff
@@ -25,47 +25,47 @@ definition {
...
+			Navigator.openSpecificURL(url = "http://localhost:8080/en/group/control_panel/manage;'*prompt()*'?p_p_id=com_liferay_my_account_web_portlet_MyAccountPortlet&p_p_lifecycle=0&p_p_state=maximized&p_p_auth=${p_p_authParam}");
 
-		AssertAlertNotPresent();
+			AssertAlertNotPresent();
+		}
 	}
 
 	@description = "This is a use case for LRQA-7595. Validade XSS vulnerabilities"
 	@priority = "4"
 	test ValidateXSSVulnerabilities {
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase:49]: Please revert this and keep the :

portalURL param usage: 

`	var portalURL = PropsUtil.get("portal.url");`

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase [hunk 4]
```diff
@@ -546,46 +519,36 @@ definition {
...
+				groupName = "Guest",
+				layoutName = "Web Content Display Page",
+				widgetName = "Web Content Display");
+		}
 
-			PageEditor.addWidget(
-				portletName = "Web Content Display",
-				targetFragmentName = "Container");
+		task ("Select web content and assert no XSS pop up") {
+			Navigator.openSpecificURL(url = "http://localhost:8080/web-content-display-page");
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/xss/XSS.testcase:543]: Please use the PortalURL param here also due, we don't know what Portal host URL will be:
See test results: https://testray.liferay.com/reports/production/logs/2021-11/test-1-11/test-portal-acceptance-pullrequest(master)/9217/functional-tomcat90-mysql57-jdk8/6/23/LocalFile.XSS_ViewWebContent/summary.html.gz

**Review by gaborlovasdotcom — CHANGES_REQUESTED** [review]
Please rever the two tests' hard coded Portal URLs, and use the Props.util instead.
Other tests are ok.

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:security:nocompile

**Review by gaborlovasdotcom — CHANGES_REQUESTED** [review]
Hi @Kvale1733 , could you please send LRQA-70508 in a separate PR?
Other two looks ok and ready to forward

**Comment by Kvale1733** [comment]
Removed LRQA-70508 from PR

**Comment by gaborlovasdotcom** [comment]
ci:test:security:nocompile

**Review by gaborlovasdotcom — APPROVED** [review]
LGTM

**Comment by gaborlovasdotcom** [comment]
ci:forward

**Pull request closed without merging**